azkaban-web-server/src/main/java/cloudflow/services/ExecutionServiceImpl.java

```diff
@@ -192,10 +200,129 @@ public JobExecution getJobExecution(String executionId, String jobDefinitionId,
+      throw new CloudFlowValidationException("Flow version must be a positive number.");
+    }
+    executableFlow.setFlowVersion(executionParameters.getFlowVersion());
+    executableFlow.setDescription(executionParameters.getDescription());
+    if (!executionParameters.getExperimentId().isEmpty()) {
+      try {
+        executableFlow.setExperimentId(Integer.parseInt(executionParameters.getExperimentId()));
+        // TODO ypadron: validate existence of experiment id
+      } catch (NumberFormatException e) {
+        throw new CloudFlowValidationException("Experiment id must be an integer.");
+      }
+    }
+    executableFlow.setSubmitUser(executionParameters.getSubmitUser());
+
+    ExecutionOptions executionOptions = new ExecutionOptions();
+    executionOptions.setFailureAction(ExecutionOptions.FailureAction.valueOf(
+        executionParameters.getFailureAction().toString()));
+
+    executionOptions.setNotifyOnFirstFailure(executionParameters.isNotifyOnFirstFailure());
+    executionOptions.setNotifyOnLastFailure(executionParameters.isNotifyFailureOnExecutionComplete());
+
+    executionOptions.setConcurrentOption(executionParameters.getConcurrentOption().getName());
+    setRuntimeProperties(executionParameters, executionOptions);
+
+    executableFlow.setExecutionOptions(executionOptions);
+  }
+
+  private void setRuntimeProperties(ExecutionParameters executionParameters,
+      ExecutionOptions executionOptions) {
+
+    if (executionParameters.getProperties().size() == 0) {
+      return;
+    }
+    // supporting only flow level properties overwrite for the POC
+    if(executionParameters.getProperties().size() > 1 ||
+        !executionParameters.getProperties().containsKey("root")) {
+      logger.error("Attempt to overwrite properties of nodes other than the root flow detected: "
+          + executionParameters.getProperties());
+      throw new CloudFlowNotImplementedException("Overwriting properties of jobs or nested flows is not "
+          + "yet supported.");
+    }
+
+    Map<String, Object> runtimeProps = executionParameters.getProperties().get("root");
+    if(runtimeProps.containsKey(CommonJobProperties.FAILURE_EMAILS)) {
+      Object propValue = runtimeProps.get(CommonJobProperties.FAILURE_EMAILS);
+      List<String> emails = getNotificationEmails(propValue);
+      executionOptions.setFailureEmails(emails);
+      executionOptions.setFailureEmailsOverridden(true);
+      runtimeProps.remove(CommonJobProperties.FAILURE_EMAILS);
+    }
+
+    if(runtimeProps.containsKey(CommonJobProperties.SUCCESS_EMAILS)) {
+      Object propValue = runtimeProps.get(CommonJobProperties.SUCCESS_EMAILS);
+      List<String> emails = getNotificationEmails(propValue);
+      executionOptions.setSuccessEmails(emails);
+      executionOptions.setSuccessEmailsOverridden(true);
+      runtimeProps.remove(CommonJobProperties.SUCCESS_EMAILS);
+    }
+
+    Map<String,String> AzRuntimeProps = runtimeProps.entrySet().stream()
```

Comment:
Nit: identifier should begin with lowercase

Reply:
yes, thanks for catching this typo.